packages/salesforce/src/meta/metadata.js

```diff
@@ -0,0 +1,49 @@
+import salesforceHelper from './helper.js';
+import { createEntity, createMock } from '@openfn/metadata';
+
+const metadata = async (configuration = {}, mock = false) => {
+  let helper = await salesforceHelper(configuration);
+  if (mock) {
+    helper = createMock(helper);
+  }
+
+  // get the globals
+  const globals = await helper.getGlobals();
+
+  const root = createEntity('salesforce', 'model');
+
+  // build a model of each global sobject
+  await Promise.all(
+    globals.map(
+      ({ name, custom }) =>
+        new Promise(async resolve => {
+          const props = {};
+          if (!custom) {
+            props.system = true;
+          }
+          const e = createEntity(name, 'sobject', props);
+
+          const fields = await helper.getFields(name);
+          if (fields) {
+            fields.forEach(({ name, type, label, externalId }) => {
+              const f = createEntity(name, 'field', {
+                datatype: type,
+                label,
+                externalId,
+              });
+              e.addChild(f);
+
+              // Now the field's attributes
+            });
+
+            root.addChild(e);
+          }
+          resolve();
```

Comment:
So I don't think the adaptor metadata functions should do much, if any, error handling. That's up to whoever is writing the function. But as a rule, it's appropriate for these functions to just throw an exception if something goes wrong.

The CLI has a bit more responsibility for error handling - what does the CLI do when a metadata function throws?

The answer right now is not much but it will at least catch and exit the process with an error code, which is probably OK for now